### ISSUE TITLE
FFMPEG RTMP sync issue

### ISSUE THREAD

**Issue by sahilupadhye** [microsoftgraph/microsoft-graph-comms-samples#491]
Hi, I am trying to record the Audio and Video stream for audit purpose

I am using ffmpeg command line to serve my requirement
The video format is H264 and audio format is PCM 44.1
command I am using is 
-thread_queue_size 512 -y -i \\.\pipe\videopipe -f u16le -ar 44100 -ac 2 -i \\.\pipe\audiopipe -c:v libx264 -c:a aac -preset ultrafast -f flv -flvflags no_duration_filesize rtmp://<azuremediaserviceurl>

I am directly writing the audio and video bytes to the named pipes as I get them.
I am able to push the streams to azure media service but the audio video is out of sync.

Can the timestamps help to solve the sync issue ?
Can you please help me correct the command line params to fix the sync issue ?

**Comment by 1fabi0** [comment]
Yes timestamps can help to solve this issue but you could also try to switch to 16khz audio, because 44.1 khz is recoded on the client from 16khz which may take some time and creates a bit more out of sync.

**Comment by mosoftwareenterprises** [comment]
@sahilupadhye do you get a resolution to this?
Can the "-use_wallclock_as_timestamps 1" setting of ffmpeg help?
